### ISSUE TITLE
Extensions keep getting enabled & installed again after disabling them globally or uninstalled

### ISSUE THREAD

**Issue by Yash7200** [microsoft/vscode#233272]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.95.1 (System Setup)
- OS Version: Windows 11 Home (24H2 OS Build 26100.2161)

Steps to Reproduce:

1.  I made clean install of both Windows 11 & VS Code (first installed using user setup then uninstalled due to issue & reinstalled using System Setup, currently using vs code from system installer)
2. After installing vs code I signed into my older Microsoft Account & GitHub account.
3. After this my settings & extensions previously used on Windows 10 started installing (there was no issue with vs code earlier)

4. I wanted to _**disable few extensions but they got enabled again**_ automatically & there're extensions disabled since sync so from them _**I wanted to enable them but they don't get enabled if I try.**_
5. I tried to uninstall extensions but they got reinstalled automatically.

6. I tried to resolve this issue from my end by uninstalling vs code & reinstalling it but it didn't make any difference.
7. How can I solve this 2 issues  (1. unable to enable/disable, 2. unable to uninstall)

**Comment by ntaraujo** [comment]
I'm facing the same. Cannot disable/uninstall any extension if Settings Sync is ON.

**Comment by Yash7200** [comment]
I also installed update received via automatic downloads, still issue persists

**Comment by sandy081** [comment]
> After installing vs code I signed into my older Microsoft Account & GitHub account.

I assume you did this using Back Up and Sync Settings action?

If so, try to reproduce the issue (say uninstall the extension and wait until extension gets installed) and share following log?

- F1 > Open View... > Settings Sync
- F1 > Open View... > Shared

**Comment by Yash7200** [comment]
2024-11-13 11:50:35.545 [info] Extensions: Migrating last sync user data. Resource does not exist.
2024-11-13 11:50:35.546 [info] Extensions: Last sync data state does not exist.
2024-11-13 11:50:37.559 [info] Auto Sync: Triggered by Activity
2024-11-13 11:50:37.560 [info] Sync started.
2024-11-13 11:50:37.824 [info] Settings: No changes found during synchronizing settings.
2024-11-13 11:50:37.827 [info] Keybindings: No changes found during synchronizing keybindings.
2024-11-13 11:50:37.831 [info] Snippets: No changes found during synchronizing snippets.
2024-11-13 11:50:37.833 [info] Tasks: No changes found during synchronizing tasks.
2024-11-13 11:50:37.836 [info] GlobalState: No changes found during synchronizing ui state.
2024-11-13 11:50:37.837 [info] Extensions: Migrating last sync user data. Resource does not exist.
2024-11-13 11:50:37.837 [info] Extensions: Last sync data state does not exist.
2024-11-13 11:50:38.789 [info] Extensions: Skipped synchronizing extension because the extension is not found. coenraads.bracket-pair-colorizer
2024-11-13 11:50:38.789 [info] Extensions: Skipped synchronizing extension because the extension is not found. codemooseus.vscode-devtools-for-chrome
2024-11-13 11:50:38.790 [info] Extensions: Skipped synchronizing extension because the extension is not found. mwpb.java-prettier-formatter
2024-11-13 11:50:38.804 [info] Extensions: Skipped synchronizing extension because the extension is not found. ms-vsliveshare.vsliveshare-pack
2024-11-13 11:50:38.984 [info] Extensions: Skipped synchronizing extension because the extension is not found. vsciot-vscode.vscode-arduino
2024-11-13 11:50:48.193 [error] Invalid: Extension is not signed., Cannot install 'amodio.find-related' extension because of manifest mismatch with Marketplace
    at Jl.M (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:70:17594)
    at async Jl.I (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:70:13513)
    at async Jl.installGalleryExtensions (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:70:8893)
    at async va.updateLocalExtensions (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:98:61139)
    at async Lh.tb (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:98:55931)
    at async Lh.ab (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:98:37216)
    at async Lh.Z (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:98:35592)
    at async Lh.Y (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:98:34632)
    at async Lh.U (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:98:33239)
    at async Lh.sync (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:98:32536)
    at async $n.sync (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:98:132088)
    at async eu.S (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:98:124037)
    at async eu.P (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:98:122840)
2024-11-13 11:50:48.194 [error] extensions: Extension is not signed., Cannot install 'amodio.find-related' extension because of manifest mismatch with Marketplace
2024-11-13 11:50:48.195 [info] Profiles: No changes found during synchronizing profiles.
2024-11-13 11:50:48.197 [info] Sync done. Took 10640ms
2024-11-13 11:50:48.255 [info] Extensions: Migrating last sync user data. Resource does not exist.
2024-11-13 11:50:48.255 [info] Extensions: Last sync data state does not exist.
2024-11-13 11:50:50.258 [info] Auto Sync: Triggered by Activity
2024-11-13 11:50:50.258 [info] Sync started.
2024-11-13 11:50:50.514 [info] Settings: No changes found during synchronizing settings.
2024-11-13 11:50:50.519 [info] Keybindings: No changes found during synchronizing keybindings.
2024-11-13 11:50:50.522 [info] Snippets: No changes found during synchronizing snippets.
2024-11-13 11:50:50.525 [info] Tasks: No changes found during synchronizing tasks.
2024-11-13 11:50:50.528 [info] GlobalState: No changes found during synchronizing ui state.
2024-11-13 11:50:50.529 [info] Extensions: Migrating last sync user data. Resource does not exist.
2024-11-13 11:50:50.529 [info] Extensions: Last sync data state does not exist.
2024-11-13 11:50:51.506 [info] Extensions: Skipped synchronizing extension because the extension is not found. codemooseus.vscode-devtools-for-chrome
2024-11-13 11:50:51.507 [info] Extensions: Skipped synchronizing extension because the extension is not found. ms-vsliveshare.vsliveshare-pack
2024-11-13 11:50:51.507 [info] Extensions: Skipped synchronizing extension because the extension is not found. coenraads.bracket-pair-colorizer
2024-11-13 11:50:51.508 [info] Extensions: Skipped synchronizing extension because the extension is not found. vsciot-vscode.vscode-arduino
2024-11-13 11:50:51.509 [info] Extensions: Skipped synchronizing extension because the extension is not found. mwpb.java-prettier-formatter
2024-11-13 11:50:57.958 [error] Invalid: Extension is not signed., Cannot install 'amodio.find-related' extension because of manifest mismatch with Marketplace
    at Jl.M (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:70:17594)
    at async Jl.I (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:70:13513)
    at async Jl.installGalleryExtensions (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:70:8893)
    at async va.updateLocalExtensions (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:98:61139)
    at async Lh.tb (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:98:55931)
    at async Lh.ab (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:98:37216)
    at async Lh.Z (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:98:35592)
    at async Lh.Y (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:98:34632)
    at async Lh.U (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:98:33239)
    at async Lh.sync (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:98:32536)
    at async $n.sync (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:98:132088)
    at async eu.S (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:98:124037)
    at async eu.P (file:///c:/Program%20Files/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-utility/sharedProcess/sharedProcessMain.js:98:122840)
2024-11-13 11:50:57.960 [error] extensions: Extension is not signed., Cannot install 'amodio.find-related' extension because of manifest mismatch with Marketplace
2024-11-13 11:50:57.961 [info] Profiles: No changes found during synchronizing profiles.
2024-11-13 11:50:57.963 [info] Sync done. Took 7708ms

**Comment by sandy081** [comment]
It will be helpful to share the action you have performed so that I can look into that specific details in the logs. Please tell me what you did and what you expected and what actually happened during this.

**Comment by Yash7200** [comment]
What I did:
- Uninstalled extension named "OneDrive Browser (By Microsoft)"

What happened:
- Same extension got reinstalled automatically

What I expected:
- the extension will get uninstalled

**Comment by Yash7200** [comment]
@sandy081 what is causing this issue when settings sync is  enabled?

**Comment by Yash7200** [comment]
@sandy081 what are the steps to resolve this bug/issue?

**Comment by sandy081** [comment]
Can you please try our [insiders](https://code.visualstudio.com/insiders) and see if this issue is resolved for you?

**Comment by Yash7200** [comment]
@sandy081 

> Can you please try our [insiders](https://code.visualstudio.com/insiders) and see if this issue is resolved for you?
I tried the same but when I attempted sign-in into my accounts (both Microsoft & GitHub) via 1st Insiders then Stable what I noticed is that my data is not being synced which includes settings, themes, shortcuts, extensions, etc.

**Comment by sandy081** [comment]
Hmm, may be please wait until the next stable release where the fix is included.

**Comment by Yash7200** [comment]
@sandy081 Today, I tried signing into VS Code Insiders (stable) & did it successfully. It was working as expected (no issues of extension) & now what I noticed in current VS Code (not insiders, its regular/LTS) is that it is trying to match cloud state of extension & doesn't allow user to modify that and I disabled a few extension in Insiders & the same extensions got disabled in Regular or LTS.

Now, I am waiting for stable release. I wrote this to inform you it works as expected
